### DISCUSSION TITLE
How do I actually operate my HP

### DISCUSSION THREAD

**Discussion by stomko11** [Arnold-n/P1P2MQTT#96] [General]
I am starting this discussion as I am failing to understand how to actually operate my heatpump. I am interested in 3 things - pump ON/OFF, DHW ON/OFF and booster heater ON/OFF. I have flashed the latest available firmware on the ESP/Arduino. I am seeing all the values, but can't figure out how to operate mentioned functions.

I have read forums, tried variety of things, such as publishing "Z1" or "Z0" to P1P2/W topic, or tried what I was advised earlier - publish "E 35 2F 1" or "E 35 2F 0" to turn on/off HP. But none of this actually changes anything.

My unit is EHBH11CB9W

Can someone advise if and what I am doing wrong?

**Comment by Arnold-n** [comment]
Hi @stomko11,
The instructions only work in "L1" mode, do you have that working?
In topic P1P2/S/# or via telnet you should see status and error messages which may help you understand what is going on.
This week I"ll release new firmware v0.9.45 to enable control from HA. In that new firmware (v0.9.45) the "simpler" commands like Z1 and Z0 are no longer supported, but in v0.9.43 these should still work.

**Comment by stomko11** [comment]
Indeed that is the issue. But how do I change it to "L1" mode?
```
P1P2/S/053 * [ESP] Uptime 276550 at 20240505_19:42:10
P1P2/S/053 * [ESP] Uptime 276560 at 20240505_19:42:20
P1P2/S/053 * [ESP] To ATmega: ->Z1<-
P1P2/S/053 * [MON] 2024-05-05_19:42:21 Command requires operation as auxiliary controller (L1)
P1P2/S/053 * [ESP] Uptime 276570 at 20240505_19:42:30
```

**Comment by Arnold-n** [comment]
With the L1 command via topic P1P2/W, just like Z1. This requires a free auxiliary slot, the interface will complain if no free slot is available. This command needs to be given only once, state will be stored in EEPROM. In v0.9.45 there will be a switch in HA to switch "L1" mode on and off.

**Comment by stomko11** [comment]
Appreciate the response here. So i issued L1:
```
P1P2/S/053 * [ESP] To ATmega: ->L1<-
P1P2/S/053 * [MON] 2024-05-05_20:08:28 Control_ID 0xF0 supported, no auxiliary controller found for 0xF0, switching control functionality on 0xF0 on
P1P2/S/053 * [MON] 2024-05-05_20:08:28 Control_id set to 0xF0
```

And after that Z1:
```
P1P2/S/053 * [ESP] To ATmega: ->Z1<-
P1P2/S/053 * [MON] 2024-05-05_20:09:09 Param35 0x0031 will be set to 0x01
P1P2/S/053 * [MON] 2024-05-05_20:09:09 Executing Z command
```

5 minutes later, heating is still OFF. Am I being impatient?

**Reply by Arnold-n** [reply]
It should switch on immediately, but depending on operating mode and system it may (also) need a write to parameter 0x002F ("E35002F01"). A short pause between both writes is needed (no buffering). In v0.9.45 both commands can be given at once in one instruction ("E35002F01 35003101").

**Reply by stomko11** [reply]
Should I just try to publish E35002F01 to P1P2/W then?

**Reply by Arnold-n** [reply]
Indeed, or both E35002F01 followed by Z1. Or wait for v0.9.45 if you are using Home Assistant.

**Comment by stomko11** [comment]
Thanks. I guess I will just wait. I am indeed using HA.

**Reply by Arnold-n** [reply]
@stomko11 full HA control was just released. It has lots of changes, please let me know if you encounter any issues.

**Reply by stomko11** [reply]
Thanks a lot. I have tested DHW and heating ON/OFF functions and they work fine!
Just out of curiosity, are you planning on documenting what some of config options are for?

**Reply by Arnold-n** [reply]
Good suggestion, thanks, I added a section to [the document how to set up HomeAssistant](https://github.com/Arnold-n/P1P2MQTT/blob/main/doc/HomeAssistant.md) to explain all controls and entities. Please let me know if you are missing anything.
